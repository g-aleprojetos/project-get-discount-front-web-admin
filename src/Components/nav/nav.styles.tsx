import styled from 'styled-components';
import colors from 'resources/colors';
import logo from 'assets/icon/logo.svg';

export const Container = styled.nav`
  display: flex;
  justify-content: space-between;
  width: 100%;
`;

export const ContainerLogo = styled.div`
  position: static;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 89px;
  color: ${colors.white};
  padding: 0 8px 0 89px;
`;

export const Logo = styled.div`
  width: 80px;
  height: 80px;
  background-image: url(${logo});
  background-repeat: no-repeat;
`;
