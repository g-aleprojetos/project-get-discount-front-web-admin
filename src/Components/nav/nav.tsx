import React from 'react';
import {Menu} from 'Components/menu';
import * as S from './nav.styles';

export type Props = {
  handleLogout: () => void;
};

export const Nav = (props: Props) => {
  const {handleLogout} = props;
  return (
    <S.Container data-testid={'test_nav'}>
      <S.ContainerLogo>
        <S.Logo data-testid={'test_logo'} />
      </S.ContainerLogo>
      <Menu handleLogout={handleLogout} />
    </S.Container>
  );
};
