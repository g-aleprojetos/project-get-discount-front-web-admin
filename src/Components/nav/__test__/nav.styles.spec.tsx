import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../nav.styles';

describe('Nav.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Container />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerLogo DEVE ser igual ao snapshot', () => {
    const containerLogo = renderer.create(<S.ContainerLogo />).toJSON();

    expect(containerLogo).toMatchSnapshot();
  });

  test('Logo DEVE ser igual ao snapshot', () => {
    const logo = renderer.create(<S.Logo />).toJSON();

    expect(logo).toMatchSnapshot();
  });
});
