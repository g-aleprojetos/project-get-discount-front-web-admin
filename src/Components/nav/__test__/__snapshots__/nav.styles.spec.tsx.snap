// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

<nav
  className="c0"
/>
`;

exports[`Nav.styles ContainerLogo DEVE ser igual ao snapshot 1`] = `
.c0 {
  position: static;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 89px;
  color: #FFFFFF;
  padding: 0 8px 0 89px;
}

<div
  className="c0"
/>
`;

exports[`Nav.styles Logo DEVE ser igual ao snapshot 1`] = `
.c0 {
  width: 80px;
  height: 80px;
  background-image: url(logo.svg);
  background-repeat: no-repeat;
}

<div
  className="c0"
/>
`;
